packages/server/src/provisioning/scim.ts

```diff
@@ -0,0 +1,175 @@
+        if (!newUser.externalId) {
+            return "User must contain externalId";
+        }
+
+        if (!newUser.email) {
+            return "User must contain email";
+        }
+
+        if (!newUser.name.formatted) {
+            return "User must contain name.formatted";
+        }
+
+        if (newUser.meta.resourceType !== "User") {
+            return 'User meta.resourceType must be "User"';
+        }
+
+        return null;
+    }
+
+    private _getDataFromScimRequest(httpReq: IncomingMessage) {
+        const url = new URL(`http://localhost${httpReq.url || ""}`);
+        const secretToken = url.searchParams.get("token") || "";
+        const orgId = url.searchParams.get("org") || "";
+
+        return {
+            secretToken,
+            orgId,
+        };
+    }
+
+    private async _handleScimUsersPost(httpReq: IncomingMessage, httpRes: ServerResponse) {
+        let newUser: ScimUserRequestData;
+
+        const { secretToken, orgId } = this._getDataFromScimRequest(httpReq);
+
+        if (!secretToken || !orgId) {
+            httpRes.statusCode = 400;
+            httpRes.end("Empty SCIM Secret Token / Org Id");
+            return;
+        }
+
+        try {
+            const body = await readBody(httpReq);
+            newUser = JSON.parse(body);
+        } catch (e) {
+            httpRes.statusCode = 400;
+            httpRes.end("Failed to read request body.");
+            return;
+        }
+
+        const validationError = this._validateScimUser(newUser);
+        if (validationError) {
+            httpRes.statusCode = 400;
+            httpRes.end(validationError);
+            return;
+        }
+
+        try {
+            const provisioning = await this.getProvisioning({ email: newUser.email });
+            const orgProvisioning = provisioning.orgs.find((org) => org.id === orgId);
+            const org = await this.storage.get(Org, orgId);
+
+            if (!orgProvisioning || !org) {
+                throw new Error("Organization not found");
+            }
+
+            // TODO: remove this once the secret is stored in the org
+            // if (secretToken !== orgProvisioning.scimSecret) {
+            if (secretToken !== "asdrtyghj") {
+                throw new Error("Invalid SCIM Secret Token");
+            }
+
+            const existingMember = orgProvisioning.members.find((member) => member.email === newUser.email);
+
+            if (existingMember) {
+                throw new Error("Member already exists");
+            }
+
+            const newProvisioningMember = new MemberProvisioning();
+            newProvisioningMember.email = newUser.email;
+
+            provisioning.account.status = newUser.active ? ProvisioningStatus.Active : ProvisioningStatus.Suspended;
+
+            orgProvisioning.members.push(newProvisioningMember);
+
+            // TODO: Save
+            // await this.storage.save(orgProvisioning);
+            // await this.storage.save(provisioning);
```

Comment:
@MaKleSoft I suppose this would be enough (didn't want to save yet as then I'll have to delete data on every test)